packages/core/test/guards.test.ts

```diff
@@ -555,6 +555,70 @@ describe('referencing guards', () => {
       Guard 'missing-predicate' is not implemented.'."
     `);
   });
+
+  it('should be possible to reference a composite guard that only uses inline predicates', () => {
+    const machine = createMachine(
+      {
+        initial: 'a',
+        states: {
+          a: {
+            on: {
+              EVENT: {
+                target: 'b',
+                guard: 'referenced'
+              }
+            }
+          },
+          b: {}
+        }
+      },
+      {
+        guards: {
+          referenced: not(() => false)
+        }
+      }
+    );
+
+    const actorRef = interpret(machine).start();
+    actorRef.send({ type: 'EVENT' });
+
+    expect(actorRef.getSnapshot().matches('b')).toBeTruthy();
+  });
+
+  it('should be possible to reference a composite guard that references other guards recursively', () => {
```

Comment:
@davidkpiano you noted down in a code comment that we should handle things recursively but I pushed out some new tests here and this things seems to work. Am I missing some cases that you were thinking about?

Reply:
This still only goes one level deep; try something like:

```ts
guards: {
  ref1: { type: 'ref2' },
  ref2: { type: 'ref3' },
  ref3: () => true
}
```